src/rgw/rgw_rest_role.h

```diff
@@ -0,0 +1,116 @@
+#ifndef CEPH_RGW_REST_ROLE_H
+#define CEPH_RGW_REST_ROLE_H
+
+class RGWRestRole : public RGWOp {
+protected:
+  string role_name;
+  string role_path;
+  string trust_policy;
+  string policy_name;
+  string perm_policy;
+  string path_prefix;
+
+public:
+  RGWRestRole() = default;
+  virtual void init(RGWRados *store, struct req_state *s, RGWHandler *h) {
+    RGWOp::init(store, s, h);
+  }
+  virtual void send_response() override;
+};
+
+class RGWRoleRead : public RGWRestRole {
+public:
+  RGWRoleRead() = default;
+  virtual int verify_permission() override;
+  virtual uint32_t op_mask() override { return RGW_OP_TYPE_READ; }
+};
+
+class RGWRoleWrite : public RGWRestRole {
+public:
+  RGWRoleWrite() = default;
+  virtual int verify_permission() override;
+  virtual uint32_t op_mask() override { return RGW_OP_TYPE_WRITE; }
+};
+
+class RGWCreateRole : public RGWRoleWrite {
+public:
+  RGWCreateRole() = default;
+  void execute() override;
+  int get_params();
```

Comment:
@cbodley : get_params() is declared and defined only in CreateRole, DeleteRole etc classes. It is not there in any of the other classes up till RGWOp.